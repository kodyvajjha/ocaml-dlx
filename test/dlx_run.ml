let () =
  let items = [ "a"; "b"; "c"; "d"; "e"; "f"; "g" ] in
  let options =
    [
      [ "c"; "e" ];
      [ "a"; "d"; "g" ];
      [ "b"; "c"; "f" ];
      [ "a"; "d"; "f" ];
      [ "b"; "g" ];
      [ "d"; "e"; "g" ];
    ]
  in
  let cur = Dlx.Algox.mk ~items ~options in
  (* CCFormat.printf "%a" Dlx.Algox.pp cur; *)
  CCFormat.printf "@.Ans = %a"
    CCFormat.Dump.(list (list string))
    (Dlx.Algox.solve_dlx cur)

let () =
  let items = [ "a"; "b"; "c"; "d"; "e"; "f"; "g"; "h"; "i" ] in
  let options =
    [
      [ "c"; "e"; "h" ];
      [ "a"; "d"; "g" ];
      [ "b"; "c"; "f" ];
      [ "a"; "d"; "f"; "i" ];
      [ "b"; "g" ];
      [ "d"; "e"; "g" ];
    ]
  in
  let cur = Dlx.Algox.mk ~items ~options in
  (* CCFormat.printf "%a" Dlx.Algox.pp cur; *)
  CCFormat.printf "@.Ans = %a"
    CCFormat.Dump.(list (list string))
    (Dlx.Algox.solve_dlx cur)
