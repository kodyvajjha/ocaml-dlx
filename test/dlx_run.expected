
Ans = [["a";"d";"f"];["b";"g"];["c";"e"]]